Dangerfile

```diff
@@ -124,6 +124,30 @@ has_app_changes.each do |file|
   end
 end
 
+# check for info field required fields
+has_app_changes.each do |file|
+  # get info field data
+  info = {}
+  content = File.read(file)
+  info_body = content.scan(/info\((.*?)\)/m)
```

Comment:
Although, this may work fine for the data that you are currently being used in the info hashes, this method of getting the info hash out of policy templates seems like it could be quite error prone if someone tries to put arbitrary strings into some value in the hash that could contain a `)`. A more robust approach would be to create a small Ruby DSL that mimics the Policy Template language but only actually implements the `info` method and uses it to collect the info hash.

Reply:
@douglaswth This is a great suggestion.  I tested this exact scenario and it caused the build to fail, because it fails to evaluate the string.  With that, I'm not concerned about this since the build fails and would force the user to correct before we'd approve..

The DSL is something I'd like to consider though.  Do you have any suggestion where to start?  An example perhaps?

Reply:
Here's a DSL that does the job as long as the `info` hash is defined before any RCL `define` declarations (I imagine everyone on the content team will be putting the metadata before any declarations).

```ruby
class DSL
  DEFINITION_REGEX = /^[[:blank:]]*define[[:blank:]]*([\w_\.]+)[[:blank:]]*\([@$\w _,]*\)[[:blank:]]*.*do/.freeze

  attr_reader :parsed_info

  def parse(file)
    source = File.read(file)

    # we assume that all of the metadata is before any declarations
    # especially RCL defines which are not parsable as Ruby
    source = source.split(DEFINITION_REGEX, 2).first

    instance_eval(source, file)
  end

  def info(**info)
    @parsed_info = info
  end

  def method_missing(symbol, *args, &block)
    # do nothing
  end
end

dsl = DSL.new
dsl.parse('path/to/policy_template.pt')

p dsl.parsed_info
```

Reply:
Thanks @douglaswth.  this is awesome.

Reply:
@douglaswth  I updated the PR to use the DSL.  I created a lib file to use in the Dangerfile and Rakefile and added other fields to parse.  so much nicer than using regex.  
Can you give another review?